### ISSUE TITLE
findOne with nested many-to-many selection 

### ISSUE THREAD

**Issue by VictorJuliani** [RobinBuschmann/sequelize-typescript#672]
**Versions**

- sequelize: 5.16.0
- sequelize-typescript: 1.0.0
- typescript: 3.4.3

**I'm submitting a ...** 

[x] bug report
[ ] feature request

**Actual behavior:**
Hi! I have a simple database mapping with this relations:
Item > hasOne > Tag
Tag > hasOne Company
Company > belongsToMany > UserCompany
User > belongsToMany > UserCompany

They seem correctly mappend using annotations and work fine when isolated.
I want to select a specific item IF it belongs to a tag that belongs to a company related to the user.
My selection is basically like this:

```
Item.findOne({
	where: { value: itemValue },
	include: [{
		model: Tag,
		required: true,
		attributes: [],
		include: [{
			model: Company,
			required: true,
			attributes: [],
			include: [{
				model: User,
				required: true,
				through: {
					where: { user_id: userId }
				}
			}]
		}]
	}]
});
```

But it throws an error of SQL syntax:
```
original: SQLError: (conn=11, no: 1054, SQLState: 42S22) Unknown column **'tag->company.id' in 'on clause'**
sql:
SELECT `ITEM`.*,
	`tag->company->users`.`id` AS `tag.company.users.id`,
	`tag->company->users->UserCompany`.`id` AS `tag.company.users.UserCompany.id`,
	`tag->company->users->UserCompany`.`user_id` AS `tag.company.users.UserCompany.user_id`,
	`tag->company->users->UserCompany`.`company_id` AS `tag.company.users.UserCompany.company_id`
FROM
	(SELECT
		`ITEM`.`id`,
		`ITEM`.`value`,
		`ITEM`.`tag_id`
	FROM `item` AS `ITEM`
	INNER JOIN `tag` AS `tag` ON `ITEM`.`tag_id` = `tag`.`id`
	INNER JOIN `company` AS `tag->company` ON `tag`.`company_id` = `tag->company`.`id`
	WHERE
		`ITEM`.`value` = '2309737967' AND
		( SELECT
			`tag`.`id`
		FROM `tag` AS `tag`
		INNER JOIN `company` AS `company` ON `tag`.`company_id` = `company`.`id`
		INNER JOIN ( `usercompany` AS `company->users->UserCompany`
			INNER JOIN `user` AS `company->users` ON `company->users`.`id` = `company->users->UserCompany`.`user_id` AND `company->users->UserCompany`.`user_id` = 1
		) ON `company`.`id` = `company->users->UserCompany`.`company_id`
		WHERE (`tag`.`id` = `ITEM`.`tag_id`) LIMIT 1 )
		IS NOT NULL LIMIT 1) AS `ITEM`
	INNER JOIN ( `usercompany` AS `tag->company->users->UserCompany`
		INNER JOIN `user` AS `tag->company->users` ON `tag->company->users`.`id` = `tag->company->users->UserCompany`.`user_id` AND `tag->company->users->UserCompany`.`user_id` = 1
	) **ON `tag->company.id` = `tag->company->users->UserCompany`.`company_id`**;
```

If I move out the _include_ User part, it works fine. Also, if I do the following query, selecting a company directly (and not as an include of another model), it works as well:

```
Company.findAll({
	include: [{
	model: User,
	required: true,
	through: {
		where: { user_id: userId }
	}
}]
});
```

**Expected behavior:**
It should load the company's users even nested

**Steps to reproduce:**
Basically recreate the same model structure and query, with a nested many-to-many relation on the query.

```ts
@Table({ tableName: 'tag', timestamps: false })
export class Tag extends Model<Tag> {
	@Column({ primaryKey: true }) id: number;
	@Column @ForeignKey(() => Company) company_id: number;
	@BelongsTo(() => Company) company: Company;
	@HasOne(() => Item) item: Item;
}

@Table({ tableName: 'item', timestamps: false })
export class Item extends Model<Item> {
	@Column({ primaryKey: true }) id: number;
	@Column value: string;
	@Column @ForeignKey(() => Tag) tag_id: number;
	@BelongsTo(() => Tag) tag: Tag;
}

@Table({ tableName: 'usercompany', timestamps: false })
export class UserCompany extends Model<UserCompany> {
	@Column({ primaryKey: true }) id: number;
	@ForeignKey(() => Company) company_id: number;
	@ForeignKey(() => User) user_id: number;
}

@Table({ tableName: 'company', timestamps: false })
export class Company extends Model<Company> {
	@Column({ primaryKey: true }) id: number;
	@BelongsToMany(() => User, () => UserCompany) users: User[];
}

@Table({ tableName: 'user', timestamps: false })
export class User extends Model<User> {
	@Column({ primaryKey: true }) id: number;
	@BelongsToMany(() => Company, () => UserCompany) companies: Company[];
}
```
```
{
  "compilerOptions": {
    "target": "es2015",
    "baseUrl": "src",
    "module": "commonjs",
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "allowJs": true,
    "importHelpers": true,
    "alwaysStrict": true,
    "sourceMap": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitReturns": true,
    "noImplicitAny": true,
    "noImplicitThis": true,
    "strictNullChecks": true,
    "suppressImplicitAnyIndexErrors": true,
    "removeComments": true,
  },
  "include": [
    "src/**/*.ts"
  ]
}
```

I have no special configs for Sequelize, only database credentials.

**Comment by RobinBuschmann** [comment]
@VictorJuliani Does piyushkantm's answer solve your issue?

**Comment by VictorJuliani** [comment]
@piyushkantm I had tried that, but without success. In fact, I tried placing `subQuery: false `everywhere and in multiple places as well. However, the outcome was always as reported.

@RobinBuschmann My solution to this was to reverse Item & Tag relation (though no changes were required in DB). Then, my selection became:

```
Tag.findOne({
	include: [{
		model: Item,
		attributes: [],
		required: true,
		where: {
			value: itemValue
		}
	}, {
		model: Company,
		required: true,
		include: [{
			model: User,
			required: true,
			through: {
				where: { user_id: userId },
				attributes: []
			}
		}]
	}],
});
```

In this way, my joins changed from Item > Tag > Company > User to Tag > Item & (Company > User).
Having this reversed relation is not a big problem to me, but it seems wrong and I still believe the original mapping was correct with a bad query output.
